manifests/istio-telemetry/kiali/templates/kiali-kialicr.yaml

```diff
@@ -0,0 +1,118 @@
+    ldap:
+{{- with .Values.kiali.dashboard.auth.strategy.ldap }}
+{{ toYaml . | indent 6 }}
+{{- end }}
+{{- end }}
+  deployment:
+{{- if .Values.kiali.accessibleNamespaces }}
+    accessible_namespaces:
+{{ toYaml .Values.kiali.accessibleNamespaces | indent 6 }}
+{{- end }}
+    affinity:
+    {{- include "nodeaffinity" . | indent 6 }}
+    {{- include "podAntiAffinity" . | indent 6 }}
+    image_pull_policy: {{ .Values.global.imagePullPolicy | default "Always" }}
+{{- if .Values.global.imagePullSecrets }}
+    image_pull_secrets:
+{{- range .Values.global.imagePullSecrets }}
+    - {{ . }}
+{{- end }}
+{{- end }}
+    ingress_enabled: {{ .Values.kiali.ingress.enabled }}
+    namespace: {{ .Release.Namespace }}
+{{- if .Values.kiali.nodeSelector }}
+    node_selector:
+{{ toYaml .Values.kiali.nodeSelector | indent 6 }}
+{{- end }}
+{{- if .Values.kiali.podAnnotations }}
+    pod_annotations:
+{{ toYaml .Values.kiali.podAnnotations | indent 6 }}
+{{- end }}
+{{- if .Values.global.priorityClassName }}
+    priority_class_name: {{ .Values.global.priorityClassName }}
+{{- end }}
+    replicas: {{ .Values.kiali.replicaCount }}
+{{- if .Values.kiali.resources }}
+    resources:
+{{ toYaml .Values.kiali.resources | indent 6 }}
+{{- end }}
+{{- if .Values.kiali.service.annotations }}
+    service_annotations:
+{{ toYaml .Values.kiali.service.annotations | indent 6 }}
+{{- end }}
+    service_type: {{ .Values.kiali.service.type }}
+    secret_name: {{ .Values.kiali.dashboard.secretName }}
+{{- if .Values.kiali.tolerations }}
+    tolerations:
+{{ toYaml .Values.kiali.tolerations | indent 6 }}
+{{- end }}
+    verbose_mode: 3
+    view_only_mode: {{ .Values.kiali.dashboard.viewOnlyMode }}
+  external_services:
+    grafana:
+      url: {{ .Values.kiali.dashboard.grafanaURL }}
+{{- if .Values.kiali.dashboard.grafanaInClusterURL}}
+      in_cluster_url: {{ .Values.kiali.dashboard.grafanaInClusterURL }}
+{{- end }}
+    istio:
+      url_service_version: http://istio-pilot.{{ .Values.global.configNamespace }}:8080/version
+    prometheus:
+{{- if .Values.global.prometheusNamespace }}
+      url: http://prometheus.{{ .Values.global.prometheusNamespace }}:9090
+{{ else }}
+      url: http://prometheus:9090
+{{- end }}
+    tracing:
+      url: {{ .Values.kiali.dashboard.jaegerURL }}
+{{- if .Values.kiali.dashboard.jaegerInClusterURL }}
+      in_cluster_url: {{ .Values.kiali.dashboard.jaegerInClusterURL }}
+{{- end }}
+{{- if .Values.kiali.security.enabled }}
+  identity: {}
+{{- else}}
+  identity:
+    cert_file: ""
+    private_key_file: ""
+{{- end}}
+{{- if .Values.kiali.excludedWorkloads }}
+  kubernetes_config:
+    excluded_workloads:
+{{ toYaml .Values.kiali.excludedWorkloads | indent 6 }}
+{{- end}}
+  login_token:
+    signing_key: {{ randAlphaNum 10 | quote }}
+  server:
+    port: 20001
+{{- if .Values.kiali.contextPath }}
+    web_root: {{ .Values.kiali.contextPath }}
+{{- end }}
+{{- end }}
+{{- end }}
```

Comment:
the major problem I see with this CR is the consumption of a bunch of key/value pairs. We just had a long discussion in the last env WG meeting that we essentially want to "start over" with a new API with minimal options and choice.

This does not mean that Kiali must also have minimal choice, as Kiali is an upstream of Istio. Instead it means that as a downstream, we may choose to limit config option choices in our direct integration of this code. Perhaps extensions and telemetry should have been called `IET` :grin:

One of the above comments I left in the review is salient here - and that is the protobuf idea (or as discussed in the last env wg meeting, perhaps a JSON blob to support all the models. TBD. Should not block this review.

Reply:
Obviously, istioctl can install Kiali CR configured however it wants. Feel free to reduce the amount of config it adds to its own Kiali CR here, thus relying on the Kiali defaults for the rest. As for Kiali itself, it will retain these settings in case people do need to do "backdoor" customization to Kiali's config.

Reply:
We should not dictate to an upstream what config options they support or don't support. I wasn't suggesting such a hardline approach and ruled that out in my second paragraph above.

What I did say, and perhaps this was unclear, was that this PR should be a little more focused around config options that are absolutely mandatory for Kiali to function. Not being a Kiali expert, this is hard for me to judge as a reviewer.